gui/qt/main_window.py

```diff
@@ -343,6 +353,8 @@ def show_or_hide(self):
 
     def bring_to_top(self):
         self.show()
+        # Make sure the window is not minimized
```

Comment:
From the Qt docs -- they say the new window state on some platforms doesn't take effect until after a window show event and that it may be delayed. https://doc.qt.io/qt-5/qwidget.html#setWindowState

I wonder if that's what's causing weirdness on MacOS?